src/snowcli/cli/streamlit/manager.py

```diff
@@ -0,0 +1,168 @@
+                )
+            standard_page_name = self.get_standard_stage_name(full_stage_name)
+            from_stage_command = f"FROM {standard_page_name}"
+        else:
+            from_stage_command = ""
+        main_file = (
+            "streamlit_app_launcher.py" if use_packaging_workaround else file.name
+        )
+
+        return self._execute_query(
+            f"""
+            create streamlit {streamlit_name}
+            {from_stage_command}
+            MAIN_FILE = '{main_file}'
+            QUERY_WAREHOUSE = {connection_ctx.warehouse};
+
+            alter streamlit {streamlit_name} checkout;
+        """
+        )
+
+    def share(self, streamlit_name: str, to_role: str) -> SnowflakeCursor:
+        return self._execute_query(
+            f"grant usage on streamlit {streamlit_name} to role {to_role}"
+        )
+
+    def drop(self, streamlit_name: str) -> SnowflakeCursor:
+        return self._execute_query(f"drop streamlit {streamlit_name}")
+
+    def deploy(
+        self,
+        streamlit_name: str,
+        file: Path,
+        open_: bool,
+        use_packaging_workaround: bool,
+        packaging_workaround_includes_content: bool,
+        pypi_download: str,
+        check_anaconda_for_pypi_deps: bool,
+        package_native_libraries: str,
+        excluded_anaconda_deps: str,
+    ):
+        stage_manager = StageManager()
+
+        # THIS WORKAROUND HAS NOT BEEN TESTETD WITH THE NEW STREAMLIT SYNTAX
+        if use_packaging_workaround:
+            # package an app.zip file, same as the other snowpark_containers_cmds package commands
+            snowpark_package(
+                pypi_download,  # type: ignore[arg-type]
+                check_anaconda_for_pypi_deps,
+                package_native_libraries,  # type: ignore[arg-type]
+            )
+
+            # upload the resulting app.zip file
+            stage_name = f"{streamlit_name}_stage"
+            stage_manager.put("app.zip", stage_name, 4, True)
+            main_module = str(file).replace(".py", "")
+            file = generate_streamlit_package_wrapper(
+                stage_name=f"{streamlit_name}_stage",
+                main_module=main_module,
+                extract_zip=packaging_workaround_includes_content,
+            )
+
+            # upload the wrapper file
+            stage_manager.put(str(file), stage_name, 4, True)
+
+            # if the packaging process generated an environment.snowflake.txt
+            # file, convert it into an environment.yml file
+            excluded_anaconda_deps_list: Optional[List[str]] = None
+            if excluded_anaconda_deps is not None:
+                excluded_anaconda_deps_list = excluded_anaconda_deps.split(",")
+            env_file = generate_streamlit_environment_file(excluded_anaconda_deps_list)
+            if env_file:
+                stage_manager.put(str(env_file), stage_name, 4, True)
```

Comment:
Have you considered extracting this to method?

Reply:
Good idea, extracted